### PR TITLE
[Serializer] Fix serialization of items with groups across entities and discrimination map

### PR DESCRIPTION
**Pull request by sroze**
| Q             | A
| ------------- | ---
| Branch?       | 4.1
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #27816, #27641
| License       | MIT
| Doc PR        | ø

I believe this approach is better than the one taken in #27816. At least, it's an alternative :)

### CODE REVIEW

**Comment by sroze** [comment]
Test failures are un-related.

**Comment by voodoo-dn** [comment]
This PR works well and fix my problem, thanks.

File: src/Symfony/Component/Serializer/Normalizer/ObjectNormalizer.php [hunk 1]
```diff
@@ -142,11 +142,16 @@ protected function getAllowedAttributes($classOrObject, array $context, $attribu
             return false;
         }
 
-        if (null !== $this->classDiscriminatorResolver && null !== $discriminatorMapping = $this->classDiscriminatorResolver->getMappingForMappedObject($classOrObject)) {
-            $allowedAttributes[] = $attributesAsString ? $discriminatorMapping->getTypeProperty() : new AttributeMetadata($discriminatorMapping->getTypeProperty());
+        $class = is_object($classOrObject) ? \get_class($classOrObject) : $classOrObject;
```
Comment (stof) [inline src/Symfony/Component/Serializer/Normalizer/ObjectNormalizer.php]: `is_object` should be fully-qualified here, to benefit from the optimized opcode.

Comment (stof) [reply src/Symfony/Component/Serializer/Normalizer/ObjectNormalizer.php]: btw, shouldn't it go inside the `if (null !== $this->classDiscriminatorResolver) {` ?

Comment (sroze) [reply src/Symfony/Component/Serializer/Normalizer/ObjectNormalizer.php]: Good points, updated 👍

**Review by nicolas-grekas — COMMENTED** [review]
(failures unrelated)
(sure we don't need more tests?)

**Comment by sroze** [comment]
@nicolas-grekas the fixture update covers the use-cases I think about so I'd say no 🤔

**Review by dunglas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @sroze.

**Pull request merged**